lnd.go

```diff
@@ -150,10 +150,7 @@ type rpcListeners func() ([]*ListenerWithSignal, func(), error)
 // Main is the true entry point for lnd. This function is required since defers
 // created in the top-level scope of a main method aren't executed if os.Exit()
 // is called.
-func Main(config *Config, lisCfg ListenerCfg) error {
-	// Hook interceptor for os signals.
-	signal.Intercept()
-
+func Main(config *Config, lisCfg ListenerCfg, shutdownChan <-chan struct{}) error {
```

Comment:
Should also update the `godoc` as well with the expected behavior of this new channel.